tests/benchdnn/dnnl_common.hpp

```diff
@@ -310,7 +310,13 @@ int check_dnnl_status(dnnl_status_t status, const prb_t *prb, res_t *res) {
         case dnnl_unimplemented: {
             // Unconditionally set all Nvidia backend unimplemented cases as
             // not supported.
-            if (is_nvidia_gpu() || is_amd_gpu()) {
+            if (is_nvidia_gpu()
+                    || is_amd_gpu()
+#ifdef DNNL_SYCL_GENERIC
+                    // skip unimplemented configs for sycl impl
+                    || is_gpu()
+#endif
```

Comment:
IIRC, the expectation was generic kernels cover oneDNN features completely, thus, all unimplemented cases should be properly reported by benchdnn and addressed by developers. Has this been changed?

@mgouicem, @vpirogov

Reply:
This is indeed the long term goal. As the generic backend is still a work in progress and new features are being added to oneDNN it is useful to be able to skip some configurations that are not supported yet.

Reply:
Removed it a bit prematurely, its added back in until a conclusion is reached

Reply:
I don't see any unimplemented hits in current validation. It makes me think if there are any new unimplemented cases, they must be local to these changes. I can provide more guidance if there's a pointer on or a snapshot of what's going on.

Reply:
Without this change running the oneDNN compiled with generic vendor (sycl ref matmul used) causes benchdnn to fail with test unimplemented error when testing with runtime dims, spec constants require kernel to be compiled for the specific spec constants (specific sizes for src/dst/weights) so current approach does now support runtime dims.
We are wondering if this can should be reported as skipped by benchdnn, as it is expected.

Reply:
This sounds to me like a matmul-specific skip.
I propose to introduce `is_generic_gpu()` method identically to `is_nvidia_gpu()`, and put a check [here](https://github.com/oneapi-src/oneDNN/blob/main/tests/benchdnn/matmul/matmul.cpp#L556) (before the if() {} section).

The change in this PR will silence ALL drivers with ANY unimplemented hit which is not the desired behavior in a long term.
The practice to mark unimplemented as skipped is to figure out the minimal scope all of them fit and skip only for it. In your case is sounds like any runtime dimensions involved. You may relate to [this statement](https://github.com/oneapi-src/oneDNN/blob/main/tests/benchdnn/matmul/matmul.cpp#L585-L587) when crafting it